src/components/utils/popover/popover-inline.ts

```diff
@@ -0,0 +1,154 @@
+import { PopoverItemHtml } from './components/popover-item/popover-item-html/popover-item-html';
+import { PopoverDesktop } from './popover-desktop';
+import { css } from './popover.const';
+import { PopoverParams } from './popover.types';
+
+/**
+ * Horizontal popover that is displayed inline with the content
+ */
+export class PopoverInline extends PopoverDesktop {
+  /**
+   * Item nested popover is displayed for
+   */
+  private nestedPopoverTriggerItem: PopoverItemDefault | PopoverItemHtml | null = null;
+
+  /**
+   * Constructs the instance
+   *
+   * @param params - instance parameters
+   */
+  constructor(params: PopoverParams) {
+    super(
+      {
+        ...params,
+        class: css.popoverInline,
+      },
+      {
+        [PopoverItemType.Default]: {
+          hint: {
+            position: 'top',
+            alignment: 'center',
+          },
+        },
+        [PopoverItemType.Html]: {
+          hint: {
+            position: 'top',
+            alignment: 'center',
+          },
+        },
+      }
+    );
+
+    /**
+     * If active popover item has children, show them.
+     * This is needed to display link url text (which is displayed as a nested popover content)
+     * once you select <a> tag content in text
+     */
+    this.items
+      .forEach((item) => {
+        if (!(item instanceof PopoverItemDefault) && !(item instanceof PopoverItemHtml)) {
+          return;
+        }
+
+        if (item.hasChildren && item.isActive) {
+          this.showNestedItems(item);
+        }
+      });
+  }
+
+  /**
+   * Returns visible element offset top
+   */
+  public get offsetLeft(): number {
+    if (this.nodes.popoverContainer === null) {
+      return 0;
+    }
+
+    return this.nodes.popoverContainer.offsetLeft;
+  }
+
+  /**
+   * Open popover
+   */
+  public override show(): void {
+    /**
+     * If this is not a nested popover, set CSS variable with width of the popover
+     */
+    if (this.nestingLevel === 0) {
+      this.nodes.popover.style.setProperty(
+        '--inline-popover-width',
+        this.renderParams.size.width + 'px'
+      );
+    }
+    super.show();
+  }
+
+  /**
+   * Disable hover event handling
+   */
+  protected override handleHover(): void {
+    // Do nothing
+  }
+
+  /**
+   * Sets CSS variable with position of item near which nested popover should be displayed.
+   * Is used for correct positioning of the nested popover
+   *
+   * @param nestedPopoverEl - nested popover element
+   * @param item – item near which nested popover should be displayed
+   */
+  protected override setTriggerItemPositionProperty(
+    nestedPopoverEl: HTMLElement,
+    item: PopoverItemDefault
+  ): void {
+    const itemEl = item.getElement();
+    const itemOffsetLeft = itemEl ? itemEl.offsetLeft : 0;
+    const totalLeftOffset = this.offsetLeft + itemOffsetLeft;
+
+    nestedPopoverEl.style.setProperty(
+      '--trigger-item-left',
```

Comment:
I wonder if it would be better to put all the properties into some constants or enum?

Reply:
added enum with variables